actionpack/lib/action_dispatch/middleware/trace_to_file_extractor.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+# :markup: markdown
+
+module ActionDispatch
+  # # Extract file and line from a trace and open it in the editor
+  #
+  # This is used to open the file in the editor when the user clicks on the edit icon. You have to set the `EDITOR` environment variable to the editor you want to use.
+  class TraceToFileExtractor
```

Comment:
Maybe leave this `:nodoc:` for now, otherwise making changes to the list of known editors becomes a breaking change